File: backend/app/schemas/sage_export.py
from datetime import datetime

from pydantic import BaseModel, Field


class SageExportConfigResponse(BaseModel):
    id: str
    company_id: str
    warehouse_code: str
    export_directory: str | None = None
    column_mapping: str | None = None
    is_active: bool
    last_export_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SageExportConfigUpdate(BaseModel):
    warehouse_code: str | None = None
    export_directory: str | None = None


class SageExportRequest(BaseModel):
    date_from: datetime = Field(..., description="Start of export date range")
    date_to: datetime = Field(..., description="End of export date range")


class SageExportResponse(BaseModel):
    csv_data: str
    record_count: int
    sync_log_id: str

File: frontend/src/types/sage-export.ts
export interface SageExportConfig {
  id: string;
  company_id: string;
  warehouse_code: string;
  export_directory: string | null;
  column_mapping: string | null;
  is_active: boolean;
  last_export_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface SageExportConfigUpdate {
  warehouse_code?: string;
  export_directory?: string;
}

export interface SageExportRequest {
  date_from: string;
  date_to: string;
}

export interface SageExportResponse {
  csv_data: string;
  record_count: number;
  sync_log_id: string;
}

export interface SyncLogEntry {
  id: string;
  company_id: string;
  sync_type: string;
  source: string;
  destination: string;
  status: string;
  records_processed: number;
  records_failed: number;
  error_message: string | null;
  started_at: string;
  completed_at: string | null;
  created_at: string;
}

export interface PaginatedSyncLogs {
  items: SyncLogEntry[];
  total: number;
  page: number;
  per_page: number;
}
